app/assets/javascripts/project_find_file.js.coffee

```diff
@@ -0,0 +1,110 @@
+
+    # init key event(shortcut, find file)
+    @initKeyEvent()
+
+    # focus text input box
+    $(@element).focus()
+
+    # load file list
+    @load(@options.url)
+
+  # files pathes load
+  load: (url) ->
+    $.ajax
+      url: url
+      method: "get"
+      dataType: "json"
+      success: $.proxy((data) ->
+        $(".loading").hide()
+        @filePaths = data
+        @findFile($(@element).val())
+        @selecteRow $("tr.tree-item")
+      , this)
+
+  # find file
+  findFile: (searchTxt) ->
+      result = if searchTxt then fuzzaldrinPlus.filter(@filePaths, searchTxt) else @filePaths
+      @renderList result, searchTxt
+
+  # render result
+  renderList: (filePaths, searchTxt) ->
+    $(".tree-table > tbody").empty()
+    for filePath, i in filePaths
+      break if i > 20
+      markedFilePath = if searchTxt then highlightText filePath, searchTxt.split("") else filePath
+      blobItemUrl = "#{@options.blobUrlTemplate}/#{filePath}"
+      html = @makeHtml markedFilePath, blobItemUrl
+      $(".tree-table > tbody").append(html)
+
+  # highlight text(awefwbwgtc -> <b>a</b>wefw<b>b</b>wgt<b>c</b> )
+  highlightText = (txt, charArr) ->
+    result = ""
+    target = txt.toString()
+    for char in charArr
+      charIdx = target.toLowerCase().indexOf char.toLowerCase()
+      result += "#{target.substring 0, charIdx}#{(target.charAt charIdx).bold()}"
+      target = target.substring charIdx + 1
+    return (result + target)
+
+  # make tbody row html
+  makeHtml: (filePath, blobItemUrl) ->
+    return  @row.replace("%blobItemUrl", blobItemUrl).replace("%filePath", filePath)
+
+  # go back page
+  goBack: (event) ->
+    if event.keyCode == 27
+      event.preventdefault
+      parent.history.back()
+    return false;
+
+  # select row and focus
+  selecteRow: (rows, seletedRow, which) ->
+    if rows && rows.length > 0
+      if seletedRow && seletedRow.length > 0
+        next = null
+        # 38 up 40 down
+        if which is 38
+          next = seletedRow.prev()
+        else
+          next = seletedRow.next()
+
+        if next.length > 0
+          seletedRow.removeClass "selected"
+          seletedRow = next
+
+      else
+        seletedRow = rows.eq(0)
+      seletedRow.addClass("selected").focus()
+
+  # init key event
+  initKeyEvent: ->
+    # unbind and bind keyup/keydown event at text input box
+    $(document).off "keyup"
+    $(document).off "keyup", @element
+    $(document).off "keydown", "body"
+
+    $(document).on "keyup", @goBack
+    $(document).on "keyup", @element, (event) =>
+      keyCode = event.keyCode if event
+      # up down right left arrow(37~40) esc(27) caps(20) ctrl(17) alt(18) shift(16) enter(13) cmd(91)
+      if (keyCode < 37 || keyCode > 40) && keyCode != 20 && keyCode != 17 && keyCode != 18 && keyCode != 16 && keyCode != 13 && keyCode != 91

```

Comment:
Should we define these key codes as constants (e.g. https://github.com/cybertoothca/keyevent/blob/master/src/keyevent.js)?